### ISSUE TITLE
Can gradients be transferred between two models when using Crypten?

### ISSUE THREAD

**Issue by DemonWang555** [facebookresearch/CrypTen#411]
For help: I want to use Crypten to realize split learning. 
The process of positive feedback is that the output of model_A is the input of model_B. The output of model B will generate a gradient for back propagation, but how can model A get the gradient of model B's input layer?
For example:
```
mashed_data = model_A(input)
output = model_B(output)

loss_value = loss(output, labels)
loss_value.backward()

model_B.update_parameters(learning_rate)
```
To my knowledge, `loss_value.backward()` will execute the back propagation process of model_B. So, should there be one more 'backward()' for model A?
And the `.update_parameters(learning_rate)` will update the model_B. But what should I do with model_A?
Thx!

**Comment by lvdmaaten** [comment]
I haven't tested this but, presuming that `model_A` and `model_B` have the same model architecture, you should be able to copy the references to the `grad` tensor in each parameter tensor in `model_A` to the corresponding parameter tensor in `model_B`. So something like this should work:
```python
grads = {}
for name, param in model_A.named_parameters():
    grads[name] = param.grad
for name, param in model_B.named_parameters():
    param.grad = grads[name]
del gradients  # remove references to these gradients to enable garbage collection
```
You could add some logic to check that all parameter names indeed match between the two models, akin to the `strict` mode in `load_state_dict()`.
